src/linear_algebra/eigenspace.lean

```diff
@@ -197,104 +197,123 @@ noncomputable instance [is_alg_closed K] [finite_dimensional K V] [nontrivial V]
   inhabited f.eigenvalues :=
 ⟨⟨f.exists_eigenvalue.some, f.exists_eigenvalue.some_spec⟩⟩
 
-/-- Eigenvectors corresponding to distinct eigenvalues of a linear operator are linearly
-    independent. (Lemma 5.10 of [axler2015])
-
-    We use the eigenvalues as indexing set to ensure that there is only one eigenvector for each
-    eigenvalue in the image of `xs`. -/
-lemma eigenvectors_linear_independent (f : End K V) (μs : set K) (xs : μs → V)
-  (h_eigenvec : ∀ μ : μs, f.has_eigenvector μ (xs μ)) :
-  linear_independent K xs :=
+/-- The eigenspaces of a linear operator form an independent family of subspaces of `V`.  That is,
+any eigenspace has trivial intersection with the span of all the other eigenspaces. -/
+lemma eigenspaces_independent (f : End K V) : complete_lattice.independent f.eigenspace :=
 begin
   classical,
-  -- We need to show that if a linear combination `l` of the eigenvectors `xs` is `0`, then all
-  -- its coefficients are zero.
-  suffices : ∀ l, finsupp.total μs V K xs l = 0 → l = 0,
-  { rw linear_independent_iff,
-    apply this },
+  -- Define an operation from `Π₀ μ : K, f.eigenspace μ`, the vector space of of finitely-supported
+  -- choices of an eigenvector from each eigenspace, to `V`, by sending a collection to its sum.
+  let S : @linear_map K K _ _ (ring_hom.id K) (Π₀ μ : K, f.eigenspace μ) V
```

Comment:
This looks a lot like `direst_sum.submodule_coe`.

I suspect most of this proof is the same as `direct_sum.submodule_is_internal.independent`?

Reply:
Oh nevermind, I see the proof uses `complete_lattice.independent_iff_dfinsupp_lsum_injective` below which is basically the same statement.